embassy-rp/src/lib.rs

```diff
@@ -8,6 +8,7 @@ pub mod dma;
 pub mod gpio;
 pub mod interrupt;
 pub mod spi;
+#[cfg(feature = "time-driver")]
```

Comment:
these changes are unrelated. Making the time-driver optional sounds good to me, but could you do it in a separate PR?